src/MiddlewareListener.php

```diff
@@ -146,6 +147,19 @@ private function createPipeFromSpec(
                 throw InvalidMiddlewareException::fromMiddlewareName($middlewareName);
             }
 
+            // Decorate double-pass middleware
+            if (is_callable($middlewareToBePiped)) {
+                $r = $this->getReflectionFunction($middlewareToBePiped);
```

Comment:
Please use always meaningful names for variables. (loops are an exception)

But do we really need a reflection here? Smells like a hack.
A check of the interfaces and a simple `method_exists` is maybe enough.

Reply:
This has been taken from `zend-stratigility` as a way of finding whether the middleware is a double-pass one, but I'll change it (it seemed like an over-complication to me, too).

Reply:
> This has been taken from zend-stratigility…

Ah, good to know.

> it seemed like an over-complication to me, too

If the interface `Interop\Http\ServerMiddleware\MiddlewareInterface` is not present but the the current middleware is callable, then there is also no check for the parameters.
And the decorator `DoublePassMiddlewareDecorator` includes also test of the result.

Reply:
I've applied changes. Now we check whether the callable is missing the `process` method or if it does not implement the `Interop\Http\ServerMiddleware\MiddlewareInterface`, then wraps it, using the decorator.